ankaios_sdk/ankaios.py

```diff
@@ -0,0 +1,733 @@
+            print("Workload started successfully.")
+
+- Delete a workload:
+    .. code-block:: python
+
+        if ankaios.delete_workload(workload_name):
+            print("Workload deleted successfully.")
+
+- Get a workload:
+    .. code-block:: python
+
+        workload = ankaios.get_workload(workload_name)
+
+- Get the state:
+    .. code-block:: python
+
+        state = ankaios.get_state()
+
+- Get the agents:
+    .. code-block:: python
+
+        agents = ankaios.get_agents()
+
+- Get the workload states:
+    .. code-block:: python
+
+        workload_states = ankaios.get_workload_states()
+
+- Get the workload states on an agent:
+    .. code-block:: python
+
+        workload_states = ankaios.get_workload_states_on_agent(agent_name)
+
+- Get the workload states on a workload name:
+    .. code-block:: python
+
+        workload_states =
+            ankaios.get_workload_states_for_name(workload_name)
+"""
+
+__all__ = ["Ankaios", "AnkaiosLogLevel"]
+
+import logging
+from typing import Union
+from enum import Enum
+import threading
+from google.protobuf.internal.encoder import _VarintBytes
+from google.protobuf.internal.decoder import _DecodeVarint
+
+from ._protos import _control_api
+from ._components import Workload, CompleteState, Request, Response, \
+                         ResponseEvent, WorkloadStateCollection, Manifest
+
+
+class AnkaiosLogLevel(Enum):
+    """ Ankaios log levels. """
+    FATAL = logging.FATAL
+    "(int): Fatal log level."
+    ERROR = logging.ERROR
+    "(int): Error log level."
+    WARN = logging.WARN
+    "(int): Warning log level."
+    INFO = logging.INFO
+    "(int): Info log level."
+    DEBUG = logging.DEBUG
+    "(int): Debug log level."
+
+
+# pylint: disable=too-many-public-methods
+class Ankaios:
+    """
+    This class is used to interact with the Ankaios using an intuitive API.
+    The class automatically handles the session creation and the requests
+    and responses sent and received over the Ankaios Control Interface.
+
+    Attributes:
+        logger (logging.Logger): The logger for the Ankaios class.
+        path (str): The path to the control interface.
+    """
+    ANKAIOS_CONTROL_INTERFACE_BASE_PATH = "/run/ankaios/control_interface"
+    "(str): The base path for the Ankaios control interface."
+
+    DEFAULT_TIMEOUT = 5.0
+    "(float): The default timeout, if not manually provided."
+
+    def __init__(self) -> None:
+        """Initialize the Ankaios object."""
+        self.logger = None
+        self.path = self.ANKAIOS_CONTROL_INTERFACE_BASE_PATH
+
+        self._read_thread = None
+        self._connected = False
+        self._responses_lock = threading.Lock()
+        self._responses: dict[str, ResponseEvent] = {}
+
+        self._create_logger()
+
+    def __enter__(self) -> "Ankaios":
+        """
+        Connect to the control interface.
+
+        Returns:
+            Ankaios: The Ankaios object.
+        """
+        self._connect()
```

Comment:
Almost all the methods require that the user is connected, and in case he is not, an error is raised.
Having the connect and disconnect separately, and also with the `with` syntax, looks more sugestive to the user that yes, you can create the object, sure, but you need to connect to the pipe to work.

Having it in init, the user might want to create the object and get connection errors, because the constructor, from my point of view, should not have a lot of functionality: only create the object and make it as error safe as possible, and then, do stuff that might fail.

Reply:
Will be done in the next commit.

Reply:
Done.